### ISSUE TITLE
TypeScript function type declaration is throwing invalid JSDoc error when used for @param and @typedef

### ISSUE THREAD

**Issue by mohsen1** [eslint/eslint#10257]
This came up working on Webpack adding JSDoc:

https://github.com/webpack/webpack/pull/7046

**Tell us about your environment**

* **ESLint Version:** 4.19.1
* **Node Version:** 9.9
* **npm Version:** 5.6

**What parser (default, Babel-ESLint, etc.) are you using?**
**Please show your full configuration:**

See Webpack [eslintrc](https://github.com/webpack/webpack/blob/master/.eslintrc.js)

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
/**
 * A subset of Set that offers sorting functionality
 * @template T
 *
 * @typedef {(a: T, b: T) => number} SortFunction
 */
class SortableSet extends Set {}
```

**What did you expect to happen?**
No error 

**What actually happened? Please include the actual, raw output from ESLint.**

```
valid-jsdoc syntax error
```

**Comment by kaicataldo** [comment]
Apologies that we lost track of this. Has this been resolved? My guess is that Doctrine, our JSDoc parser, is having trouble parsing this.

**Comment by nzakas** [comment]
Thanks for the suggestion. As of yesterday, we have decided to officially [end-of-life JSDoc support](https://eslint.org/blog/2018/11/jsdoc-end-of-life) in ESLint. All JSDoc features are now deprecated and we won't be fixing bugs or making any further improvements to those features.

We are recommending that people transition over to use the [`eslint-plugin-jsdoc`](https://github.com/gajus/eslint-plugin-jsdoc) plugin instead of the built-in rules in ESLint.

Thanks for understanding and we apologize for any inconvenience.

**Comment by mohsen1** [comment]
The way you guys write your response feels like I paid for ESLint! ❤️

 Thank you so much for all of the work!
